src/main/java/com/atlassian/bitbucket/jenkins/internal/model/BitbucketSingleTag.java

```diff
@@ -0,0 +1,58 @@
+package com.atlassian.bitbucket.jenkins.internal.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BitbucketSingleTag {
+
+    public ArrayList<BitbucketValueProperty> values;
+
+    @JsonCreator
+    public BitbucketSingleTag(@JsonProperty("values") ArrayList<BitbucketValueProperty> values) {
+        this.values = requireNonNull(values, "values");
+
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        BitbucketSingleTag that = (BitbucketSingleTag) o;
+        return Objects.equals(getId(), that.getId()) &&
+                Objects.equals(getDisplayId(), that.getDisplayId()) &&
+                Objects.equals(getCommitterTimestamp(), that.getCommitterTimestamp()) &&
+                Objects.equals(getMessage(), that.getMessage());
+    }
+
+    public final long getCommitterTimestamp() {
+        return values.get(0).getCommitterTimestamp();
```

Comment:
I think that because we check the nullability of the object using `requireNonNull(values, "values");` this should be fine. As the object shouldn't create if we dont have at least 1 value, we also set the limit to 1, which means we should only ever have the latest commit.

I can do a `orElse` check here but i dont think it makes sense.

Reply:
What if the `values` is an empty list?

Reply:
The only case where values could be an empty list is a branch with no commits. However i dont think this is a valid state for git as there would be no jenkinsfile associated with it.  technically you can have an orphan branch but i think that comes with a host of other issues. 

If for some reason we cant find the branch (gets deleted mid call / just as build is triggered) we fall back onto non-deterministic checkout.

Reply:
This is true for the current endpoint but the client is just a generic "get tags" endpoint- if we want to use it in the future it should be able to handle whether the tag is present or not.
But I think that should be done in the client, not the data object. Given this is a data object and values is expected in all states to have 1 element, should we just store it as a single element?

Reply:
So you're saying to parse the array of "values" that come back from the json object and then store it as a single element as we only expect 1.